File: bot.js
// bot.js
// ربات آرشیوگر تلگرام به زبان جاوااسکریپت برای استفاده در Cloudflare Workers
// این فایل باید از Worker فراخوانی بشه

export async function handleRequest(update, env) {
  const TELEGRAM_TOKEN = env.TELEGRAM_BOT_TOKEN;
  const ARCHIVE_CHANNEL_ID = env.ARCHIVE_CHANNEL_ID;
  const API_URL = `https://api.telegram.org/bot${TELEGRAM_TOKEN}`;

  const msg = update.message;
  if (!msg) return { ok: true };

  // ساخت لیبل‌ها
  let labels = `👤 فرستنده: ${msg.from?.first_name || ""} ${msg.from?.last_name || ""}\n`;
  labels += `🆔 ID: ${msg.from?.id}\n`;
  labels += `⏰ تاریخ: ${new Date(msg.date * 1000).toLocaleString("fa-IR")}\n`;

  if (msg.forward_from_chat) {
    labels += `📢 منبع: ${msg.forward_from_chat.title}\n`;
  }

  // ارسال به کانال آرشیو
  if (msg.text) {
    await fetch(`${API_URL}/sendMessage`, {
      method: "POST",
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify({
        chat_id: ARCHIVE_CHANNEL_ID,
        text: `${msg.text}\n\n${labels}`
      })
    });
  } else if (msg.photo) {
    const photo = msg.photo[msg.photo.length - 1];
    await fetch(`${API_URL}/sendPhoto`, {
      method: "POST",
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify({
        chat_id: ARCHIVE_CHANNEL_ID,
        photo: photo.file_id,
        caption: labels
      })
    });
  } else if (msg.document) {
    await fetch(`${API_URL}/sendDocument`, {
      method: "POST",
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify({
        chat_id: ARCHIVE_CHANNEL_ID,
        document: msg.document.file_id,
        caption: labels
      })
    });
  } else if (msg.video) {
    await fetch(`${API_URL}/sendVideo`, {
      method: "POST",
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify({
        chat_id: ARCHIVE_CHANNEL_ID,
        video: msg.video.file_id,
        caption: labels
      })
    });
  }

  return { ok: true };
}

File: bot.py
# bot.py

import os
from telegram import Update
from telegram.ext import ApplicationBuilder, CommandHandler, MessageHandler, filters, ContextTypes
import config
from utils.labels import generate_labels
from scheduler import start_scheduler

async def start(update: Update, context: ContextTypes.DEFAULT_TYPE):
    await update.message.reply_text("ربات آرشیوگر فعال شد!")

async def archive_post(update: Update, context: ContextTypes.DEFAULT_TYPE):
    msg = update.message
    labels = generate_labels(msg)

    if msg.photo:
        await context.bot.send_photo(
            chat_id=config.CHANNEL_ID,
            photo=msg.photo[-1].file_id,
            caption=labels
        )
    elif msg.video:
        await context.bot.send_video(
            chat_id=config.CHANNEL_ID,
            video=msg.video.file_id,
            caption=labels
        )
    elif msg.document:
        await context.bot.send_document(
            chat_id=config.CHANNEL_ID,
            document=msg.document.file_id,
            caption=labels
        )
    else:
        await context.bot.send_message(
            chat_id=config.CHANNEL_ID,
            text=(msg.text or '') + "\n\n" + labels
        )

if __name__ == "__main__":
    app = ApplicationBuilder().token(config.TOKEN).build()

    # هندلرها
    app.add_handler(CommandHandler("start", start))
    app.add_handler(MessageHandler(filters.ALL, archive_post))

    # شروع Scheduler
    start_scheduler()

    # اجرای وب‌هوک
    WEBHOOK_URL = os.environ.get("WEBHOOK_URL") or f"https://your-render-service.com/{config.TOKEN}"
    PORT = int(os.environ.get("PORT", 8443))

    app.run_webhook(
        listen="0.0.0.0",
        port=PORT,
        webhook_url=WEBHOOK_URL
    )
